lametro/models.py

```diff
@@ -263,41 +264,95 @@ def upcoming_board_meeting(cls):
                           .first()
 
 
+    @staticmethod
+    def _time_ago(**kwargs):
+        '''
+        Convenience method for returning localized, negative timedeltas.
+        '''
+        return datetime.now(app_timezone) - timedelta(**kwargs)
+
+
+    @staticmethod
+    def _time_from_now(**kwargs):
+        '''
+        Convenience method for returning localized, positive timedeltas.
+        '''
+        return datetime.now(app_timezone) + timedelta(**kwargs)
+
+
     @classmethod
-    def current_meeting(cls):
+    def _potentially_current_meetings(cls):
+        '''
+        Return meetings that could be "current" – that is, meetings that are
+        scheduled to start in the last six hours, or in the next five minutes.
+
+        Fun fact: The longest Metro meeting on record is 5.38 hours long (see
+        issue #251). Hence, we check for meetings scheduled to begin up to six
+        hours ago.
+
+        Used to determine whether to check Granicus for streaming meetings.
         '''
-        Discover and return events in progress.
+        six_hours_ago = cls._time_ago(hours=6)
+        five_minutes_from_now = cls._time_from_now(minutes=5)
 
-        The maximum recorded meeting duration is 5.38 hours, according to the
-        spreadsheet provided by Metro in issue #251. So, to determine initial
-        list of possible current events, we look for all events scheduled
-        in the past 6 hours.
+        return cls.objects.filter(start_time__gte=six_hours_ago,
+                                  start_time__lte=five_minutes_from_now)\
+                           .exclude(status='cancelled')
 
-        A meeting displays as "current" if:
 
-        (1) it started in the last six hours, or it starts in the next five
-            minutes (determined by this method); and
-        (2) it started less than 55 minutes ago, and the previous meeting has
-            ended (determined by `calculate_current_meetings`); or
-        (3) Legistar indicates it is in progress (detemined by `calculate_
-            current_meetings`).
+    @classmethod
+    def _streaming_meeting(cls):
+        '''
+        Granicus provides a running events endpoint that returns an array of
+        GUIDs for streaming meetings. Metro events occur one at a time, but two
+        GUIDs appear when an event is live: one for the English audio, and one
+        for the Spanish audio.
+
+        Hit the endpoint, and return the corresponding meeting, or an empty
+        queryset.
+        '''
+        running_events = requests.get('http://metro.granicus.com/running_events.php')
+
+        for guid in running_events.json():
+            # We get back two GUIDs, but we won't know which is the English
+            # audio GUID stored in the 'guid' field of the extras dict. Thus,
+            # we iterate.
+            meeting = cls.objects.filter(extras__guid=guid.upper())
```

Comment:
seeing `guid.upper()` suggests something is pretty wrong upstream?

Reply:
guids are all-uppercase [in the api](http://webapi.legistar.com/v1/metro/events/915), but are all-lowercase in the running events endpoint. i'd prefer not to alter the source data in the scrape, so it seems appropriate to do a transformation for this instance-specific comparison. happy to hear an alternate perspective!

Reply:
yikes, okay. add a comment that explains why this thing is happening.